instrumentation/github.com/aws/aws-sdk-go/service/otels3/s3_test.go

```diff
@@ -0,0 +1,164 @@
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3iface"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"go.opentelemetry.io/contrib/instrumentation/github.com/aws/aws-sdk-go/service/otels3/mocks"
+	mockmetric "go.opentelemetry.io/contrib/internal/metric"
+	mocktrace "go.opentelemetry.io/contrib/internal/trace"
+	"go.opentelemetry.io/otel/label"
+
+	"reflect"
+	"testing"
+
+	"go.opentelemetry.io/otel/api/global"
+	"go.opentelemetry.io/otel/api/metric"
+	"go.opentelemetry.io/otel/api/trace"
+)
+
+var (
+	s3bucket = "s3bucket"
+)
+
+func getLabelValFromMeasurementBatch(key label.Key, batch mockmetric.Batch) *label.Value {
+	for _, label := range batch.Labels {
+		if label.Key == key {
+			return &label.Value
+		}
+	}
+	return nil
+}
+
+func getLabelValFromSpan(key label.Key, span mocktrace.Span) *label.Value {
+	if value, ok := span.Attributes[key]; ok {
+		return &value
+	}
+	return nil
+}
+
+func Test_instrumentedS3_PutObjectWithContext(t *testing.T) {
+	type fields struct {
+		S3API                    s3iface.S3API
+		tracer                   trace.Tracer
+		meter                    metric.Meter
+		spanCorrelationInMetrics bool
+		mockSetup                func(s3Client *mock.Mock) (expectedReturn interface{})
+	}
+	type args struct {
+		ctx   aws.Context
+		input *s3.PutObjectInput
+		opts  []request.Option
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "instrumentedS3.PutObjectWithContext should be delegated to S3.PutObjectWithContext while metrics and spans are linked",
+			fields: fields{
+				S3API:                    &mocks.S3Client{},
+				spanCorrelationInMetrics: true,
+				mockSetup: func(m *mock.Mock) (expectedReturn interface{}) {
+					expectedReturn = &s3.PutObjectOutput{}
+					m.On("PutObjectWithContext", mock.Anything, mock.Anything).Return(expectedReturn, nil)
+					return
+				},
+			},
+			args: args{
+				ctx: context.Background(),
+				input: &s3.PutObjectInput{
+					Bucket: &s3bucket,
+				},
+				opts: nil,
+			},
+			wantErr: false,
+		},
+		{
+			name: "instrumentedS3.PutObjectWithContext should be delegated to S3.PutObjectWithContext while metrics and spans are NOT linked",
+			fields: fields{
+				S3API:                    &mocks.S3Client{},
+				spanCorrelationInMetrics: false,
+				mockSetup: func(m *mock.Mock) (expectedReturn interface{}) {
+					expectedReturn = &s3.PutObjectOutput{}
+					m.On("PutObjectWithContext", mock.Anything, mock.Anything).Return(expectedReturn, nil)
+					return
+				},
+			},
+			args: args{
+				ctx: context.Background(),
+				input: &s3.PutObjectInput{
+					Bucket: &s3bucket,
+				},
+				opts: nil,
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			_, mockedTracer := mocktrace.NewTracerProviderAndTracer("github.com/aws/aws-sdk-go/aws/service/s3")
```

Comment:
should we maybe set the tracer name to a global variable at the top of this file or something?